ts/src/pro/test/Exchange/test.watchBalance.ts

```diff
@@ -1,99 +1,23 @@
-'use strict';
 
-// ----------------------------------------------------------------------------
-
-import assert from 'assert';
 import testBalance from '../../../test/Exchange/base/test.balance.js';
-import errors from '../../../base/errors.js';
-
-
-/*  ------------------------------------------------------------------------ */
-
-export default async (exchange) => {
+import testSharedMethods from '../../../test/Exchange/base/test.sharedMethods.js';
 
+async function testWatchBalance (exchange, skippedProperties, code) {
     const method = 'watchBalance';
-    const skippedProperties = {};
-
-    if (!exchange.has[method]) {
-        console.log (exchange.id, 'does not support', method, '() method');
-        return;
-    }
-
-    console.log ('watching balance...');
-
-    let now = Date.now ();
-    const ends = now + 10000;
-
+    let now = exchange.milliseconds ();
+    const ends = now + exchange.wsMethodsTestTimeoutMs;
```

Comment:
why use a tests-only property in the exchange?

Reply:
here it is being set: https://github.com/ccxt/ccxt/blob/a81f8cc8a1a672f4d0522e6e3cc52a26f0b82d68/ts/src/test/test.ts#L312
this was to avoid intrusive changes to method signatures to pass that value as argument, so setting as exchange prop seems a possible way for now. wdyt?

Reply:
I think I would rather pass it explicitly in the method's signature, it's easier to understand where it is coming from. 

Second question, I'm not sure this is the "timeout" control we need, what if the method never returns inside the  loop? it will hang "forever" right?

Reply:
1) why does it take many changes? isn't just adding a new parameter to each method?
for instance in c# I would have to explicitly define this property which is a bit odd since this is tests-only

Reply:
we could add that in `options['wsTimeout']` etc.. what would avoid defining it in c#

Reply:
or I can leave that hardcoded as it was before?

Reply:
@ttodua Question, how many changes would it require to change this
```
testWatchOrderBook (exchange, skippedProperties, symbol)
```

to this 
```
testWatchOrderBook (exchange, skippedProperties, symbol, timeOut = 20000)
```

Reply:
I am also making some other updates in transpiler.js, so you might give up this PR review for a moment, until i turn it from **draft** into ready